handsontable/src/plugins/hiddenColumns/__tests__/plugins/mergeCells.spec.js

```diff
@@ -1791,29 +1793,32 @@ describe('HiddenColumns', () => {
         colHeaders: true,
         contextMenu: true,
         hiddenColumns: {
-          columns: [1],
+          columns: [0],
           indicators: true
         },
         mergeCells: [{ row: 1, col: 1, rowspan: 1, colspan: 3 }]
       });
 
-      const mergeArea = spec().$container.find('tr:eq(2) td:eq(1)');
+      const mergeArea = spec().$container.find('tr:eq(2) td:eq(0)');
 
       // Selected 2 ranges containing together merged area.
-      selectColumns(1);
-      keyDown('ctrl');
-      selectColumns(3);
-      keyDown('ctrl');
-      selectColumns(2);
+      simulateClick(getCell(-1, 1));
+
+      keyDown('control');
+
+      simulateClick(getCell(-1, 3));
+      simulateClick(getCell(-1, 2));
+
+      keyUp('control');
```

Comment:
I would recommend to revert all the changes within that file. From my perspective it looks like you changed the tests to make it pass with new API. I responded to the issue #7290 that you have linked. I think it's not a bug exactly.

Anyway, I added `listen()` call before column selection and the original test passes. Can you check that?
```suggestion
      listen();
      selectColumns(1);
      keyDown('control');
      selectColumns(3);
      keyDown('control');
      selectColumns(2);
```

Reply:
Thank you for that comment. I've done changes within c365700c93b14d5309777f3a2668c5a073471ad7.